backend/pkg/kubeconfig/kubeconfig_test.go

```diff
@@ -67,7 +68,7 @@ func TestLoadContextsFromKubeConfigFile(t *testing.T) {
 
 		contexts, errs := kubeconfig.LoadContextsFromFile(kubeConfigFile, kubeconfig.KubeConfig)
 		require.NotEmpty(t, errs, "Expected errors for invalid auth file")
-		require.Empty(t, contexts, "Expected no contexts from invalid auth file")
+		require.NotEmpty(t, contexts, "Expected contexts from invalid auth file")
```

Comment:
I think that if the context here is missing a user or something, then we could check for that in the tests.
We do have to return all clusters to the frontend but maybe in a different PR we can add a key/value to the context to show that there is a problem.

Reply:
I reopened this, because I'm still confused about this. I still think there could be a bug/undefined issue with this function. Because the error array returned is not matched up with the returned contexts.

How do the callers know which contexts failed, and which contexts are invalid?

All the LoadContexts (and LoadContextsFromMultipleFiles, LoadContextsFromBase64String) users would also need to be changed so that all the errors are returned, and all the errors are matched up.

loadContextsFromData for example could return contexts[] and errors[] with when there is no error a nil is stored in the errors[] array. In this case the caller would need to check each error element for nil or error. Maybe it should return (contexts[], contextErrors[], error) to make the error handling clear.

Reply:
You're right, and that's a great observation. The current implementation doesn't provide a clear way to know which contexts failed or are invalid, it only logs it with the data and context name. Let me fix it